packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/table/ViewRowContextMenuEntryExecutor.java

```diff
@@ -0,0 +1,140 @@
+import org.eclipse.sirius.components.view.emf.api.IViewAQLInterpreterFactory;
+import org.springframework.stereotype.Service;
+
+/**
+ * This class is in charge of delegate row context menu entry invocation for table view DSL.
+ *
+ * @author Jerome Gout
+ */
+@Service
+public class ViewRowContextMenuEntryExecutor implements IRowContextMenuEntryExecutor {
+
+    private final ViewRepresentationDescriptionPredicate viewRepresentationDescriptionPredicate;
+
+    private final IViewRepresentationDescriptionSearchService viewRepresentationDescriptionSearchService;
+
+    private final IViewAQLInterpreterFactory aqlInterpreterFactory;
+
+    private final IFeedbackMessageService feedbackMessageService;
+
+    private final IEditService editService;
+
+    private final IObjectSearchService objectSearchService;
+
+    private final Function<EObject, UUID> idProvider = (eObject) -> UUID.nameUUIDFromBytes(EcoreUtil.getURI(eObject).toString().getBytes());
+
+    public ViewRowContextMenuEntryExecutor(ViewRepresentationDescriptionPredicate viewRepresentationDescriptionPredicate,
+            IViewRepresentationDescriptionSearchService viewRepresentationDescriptionSearchService, IViewAQLInterpreterFactory aqlInterpreterFactory, IFeedbackMessageService feedbackMessageService,
+            IEditService editService, IObjectSearchService objectSearchService) {
+        this.viewRepresentationDescriptionPredicate = Objects.requireNonNull(viewRepresentationDescriptionPredicate);
+        this.viewRepresentationDescriptionSearchService = Objects.requireNonNull(viewRepresentationDescriptionSearchService);
+        this.aqlInterpreterFactory = Objects.requireNonNull(aqlInterpreterFactory);
+        this.feedbackMessageService = Objects.requireNonNull(feedbackMessageService);
+        this.editService = Objects.requireNonNull(editService);
+        this.objectSearchService = Objects.requireNonNull(objectSearchService);
+    }
+
+    @Override
+    public boolean canExecute(TableDescription tableDescription, String tableId, String rowId, String rowMenuContextEntryId) {
+        return this.viewRepresentationDescriptionPredicate.test(tableDescription);
+    }
+
+    @Override
+    public IStatus execute(IEditingContext editingContext, TableDescription tableDescription, Table table, Line row, String rowMenuContextEntryId) {
+        var optionalTableDescription = this.viewRepresentationDescriptionSearchService
+                .findById(editingContext, tableDescription.getId())
+                .filter(org.eclipse.sirius.components.view.table.TableDescription.class::isInstance)
+                .map(org.eclipse.sirius.components.view.table.TableDescription.class::cast);
+        if (optionalTableDescription.isPresent()) {
+            var viewTableDescription = optionalTableDescription.get();
+
+            AQLInterpreter interpreter = this.aqlInterpreterFactory.createInterpreter(editingContext, (View) viewTableDescription.eContainer());
```

Comment:
Please perform a check before casting things like that. You can't know that the description will always be under a view.